scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2521,11 +2521,22 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
               val trm =
                 if (token.is[Indentation.Indent]) blockExpr()
                 else if (location != BlockStat) expr()
-                else block()
```

Comment:
should we redefine
```
  def block(): Term = autoPos {
    val res = Term.Block(blockStatSeq())
    if (token.is[LeftBrace]) inBraces(res) else res
  }
```
and just use it everywhere? for instance, line 2257 (under `Try`) would become:
```
        val body: Term = token match {
          case _ if dialect.allowTryWithAnyExpr => expr()
          case LeftParen() => inParensOrUnit(expr())
          case _ => block()
        }
```

Reply:
Might be a better idea, I will take a look at it.

Reply:
Reworked it a bit and managed to get rid of most repeated stuff.